mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp

```diff
@@ -674,7 +674,7 @@ class FtoICastOpConversion : public OpConversionPattern<CastOp> {
     Type actualResultType = dstType;
```

Comment:
I might be wrong and only real quick at this but do we need the if clause at all if `actualResultType`  is set to `dstType` here already?

Reply:
Isn't `dstType` signless? If I'm not mistaking, this condition is there so that the first cast actually goes to unsigned no matter what the signedness of `dstType` is.

Reply:
Yes, what @cferry-AMD said is also my understanding.